run_project_tests.py

```diff
@@ -731,7 +730,37 @@ def force_regenerate() -> None:
     return testresult
 
 
-def load_test_json(t: TestDef, stdout_mandatory: bool) -> T.List[TestDef]:
+# processing of test.json 'skip_*' keys, which can appear at top level, or in
+# matrix:
+def _skip_keys(test_def: T.Dict, mesonenv: environment.Environment) -> T.Tuple[bool, bool]:
+    skip_expected = False
+
+    # Test is expected to skip if MESON_CI_JOBNAME contains any of the list of
+    # substrings
+    if ('skip_on_jobname' in test_def) and (ci_jobname is not None):
+        skip_expected = any(s in ci_jobname for s in test_def['skip_on_jobname'])
+
+    # Test is expected to skip if os matches
+    if 'skip_on_os' in test_def:
+        for skip_os in test_def['skip_on_os']:
+            if skip_os.startswith('!'):
+                if mesonenv.machines.host.system != skip_os[1:]:
+                    skip_expected = True
+            else:
+                if mesonenv.machines.host.system == skip_os:
+                    skip_expected = True
```

Comment:
Isn't using a full `Environment` way too overkill here? Alternatively, wouldn't it also be possible to create a "fake" Environment without a build dir, thus eliminating the need of an `TemporaryDirectoryWinProof`, directly here?

Reply:
There's only a single Environment object made per detect_tests_to_run(), so it's not particularly heavyweight, and I think it's possible in future we might want to have some other tests in load_tests_json which require one.

(There's some refactoring which could be done to pass an Environment object around and down to stuff which detect_tests_to_run calls which needs an Environment, e.g. have_objc_compiler etc., which each make one at the moment.
The `with TemporaryDirectoryWinProof` then construct a `environment.Environment` stanza appears 4 times in this file now)

Reply:
My main problem is that it is passed through a lot of functions just that don't do anything with it, just to be used here...

So, could you try if `environment.Environment(None, None, get_fake_options('/'))` (without a temp dir) also works, so that it can be directly constructed here? If it doesn't then I guess it is fine the way it currently is.